### ISSUE TITLE
Stratosphere 0.4 cannot read from HDFS 0.20.2

### ISSUE THREAD

**Issue by fhueske** [stratosphere/stratosphere#356]
When trying to read data from HDFS 0.20.2 with Stratosphere 0.4 the following exception is thrown:

```
ERROR: The program execution failed: eu.stratosphere.nephele.executiongraph.GraphConversionException: Cannot compute input splits for CHAIN DataSource(Input Lines) -> Map(Tokenize Lines) -> Combine(Count Words): java.io.IOException: The given file URI (hdfs://host:port/my/input) described the host and port of an HDFS Namenode, but the File System could not be initialized with that address.
    at eu.stratosphere.runtime.fs.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:194)
    at eu.stratosphere.core.fs.FileSystem.get(FileSystem.java:220)
    at eu.stratosphere.core.fs.Path.getFileSystem(Path.java:293)
    at eu.stratosphere.api.common.io.FileInputFormat.createInputSplits(FileInputFormat.java:393)
    at eu.stratosphere.api.common.io.FileInputFormat.createInputSplits(FileInputFormat.java:68)
    at eu.stratosphere.pact.runtime.task.DataSourceTask.computeInputSplits(DataSourceTask.java:325)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.createVertex(ExecutionGraph.java:552)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.constructExecutionGraph(ExecutionGraph.java:273)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.<init>(ExecutionGraph.java:174)
    at eu.stratosphere.nephele.jobmanager.JobManager.submitJob(JobManager.java:551)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at eu.stratosphere.nephele.ipc.RPC$Server.call(RPC.java:416)
    at eu.stratosphere.nephele.ipc.Server$Handler.run(Server.java:944)
Caused by: java.lang.NullPointerException
    at eu.stratosphere.runtime.fs.hdfs.DistributedFileSystem.initialize(DistributedFileSystem.java:191)
    ... 15 more

    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.createVertex(ExecutionGraph.java:555)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.constructExecutionGraph(ExecutionGraph.java:273)
    at eu.stratosphere.nephele.executiongraph.ExecutionGraph.<init>(ExecutionGraph.java:174)
    at eu.stratosphere.nephele.jobmanager.JobManager.submitJob(JobManager.java:551)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
    at java.lang.reflect.Method.invoke(Method.java:597)
    at eu.stratosphere.nephele.ipc.RPC$Server.call(RPC.java:416)
    at eu.stratosphere.nephele.ipc.Server$Handler.run(Server.java:944)
```

**Comment by StephanEwen** [comment]
HDFS 0.20.2 is NOT compatible to Hadoop 1.x
HDFS 0.20.203 is.

**Comment by rmetzger** [comment]
Validated @StephanEwen's fix on Hadoop 2.2 and Hadoop 1.0.4

current 0.4-rc1 is working on yarn.

**Comment by fhueske** [comment]
Also working with HDFS 0.20.203 and 0.20.2.
Issue fixed in relaese-0.4-rc1.
